src/lsstseries/analysis/structurefunction2.py

```diff
@@ -53,9 +39,32 @@ def calc_sf2(
     executed on all available bands in `band`.
     """
 
+    if argument_container is None:
```

Comment:
We instantiate a new argument container of the correct type (based on the value of `sf_method`).